hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkMetadataUpsertPartitioner.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.action.commit;
+
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecordLocation;
+import org.apache.hudi.common.util.Option;
+
+import java.util.List;
+import java.util.Map;
+
+import scala.Tuple2;
+
+public class SparkMetadataUpsertPartitioner<T> extends SparkHoodiePartitioner<T> {
+
+  private List<BucketInfo> bucketInfoList;
+  private final int totalPartitions;
+  private Map<String, Integer> fileIdToPartitionIndexMap;
+
+  public SparkMetadataUpsertPartitioner(List<BucketInfo> bucketInfoList, Map<String, Integer> fileIdToPartitionIndexMap) {
```

Comment:
SparkMetadataTableUpsertPartitioner to assist w/ streaming writes to metadata table.